handler/main.go

```diff
@@ -18,135 +14,99 @@ import (
-	if !ok {
-		log.Fatalf("error asserting payload[\"logEvents\"].([]interface{}")
-		return
-	}
+	return handleEvents(data.LogEvents)
+}
 
+// handleEvents ... parses log messages out of log events
+func handleEvents(logEvents []events.CloudwatchLogsLogEvent) error {
 	for _, logEvent := range logEvents {
-		log.Printf("**Event (%T): %v", logEvent, logEvent)
-
-		if mStr, ok := logEvent.(map[string]interface{})["message"]; ok {
-			log.Printf("**message (%T):\n%v", mStr, mStr)
-
-			message := make(map[string]interface{})
-
-			s, ok := mStr.(string)
-			if !ok {
-				log.Fatalf("error asserting mStr.(string)")
-				return
-			}
-
-			err = json.Unmarshal([]byte(s), &message)
-			if err != nil {
-				log.Fatalf("error unmarshalling log event message: %v", err)
-				return
-			}
-
-			if eventType, ok := message["eventName"]; ok {
-				log.Printf("eventType (%T):\n%v\n", eventType, eventType)
-
-				if contains(eventTypes, eventType.(string)) {
-					log.Println("**matching eventType**")
+		var message ConsoleLoginEvent
 
-					body := makeBody(message, excludedKeys)
-					sendEmail(body)
-				}
-			}
+		err := json.Unmarshal([]byte(logEvent.Message), &message)
+		if err != nil {
+			log.Printf("error unmarshalling log event message: %v", err)
+			return err
 		}
-	}
-}
 
-func contains(s []string, str string) bool {
-	for _, a := range s {
-		if a == str {
-			return true
+		err = handleMessage(&message)
+		if err != nil {
+			return err
 		}
 	}
 
-	return false
+	return nil
 }
 
-func makeBody(m map[string]interface{}, x []string) (b string) {
-	b = "This is the header *Test*\n\n"
-	b += "EventType: " + m["eventType"].(string) + "\n"
-	b += "EventId: " + m["eventID"].(string) + "\n"
-	b += "EventTime: " + m["eventTime"].(string) + "\n"
-	b += "EventName: " + m["eventName"].(string) + "\n"
-	b += "UserAgent: " + m["userAgent"].(string) + "\n"
-	b += "AWS Region: " + m["awsRegion"].(string) + "\n"
-	b += "SourceIPAddress: " + m["sourceIPAddress"].(string) + "\n"
-
-	if !contains(x, "userIdentity") {
-		if ui, ok := m["userIdentity"].(map[string]interface{}); ok {
-			b += "\nUserIdentity \n\n"
-			b += parseSesUserIdentity(ui, x)
-		}
+// handleMessage ... filters log event messages and sends email on matches
+func handleMessage(message *ConsoleLoginEvent) error {
+	log.Printf("** " + message.EventName + " event **")
+
+	if message.EventName == "ConsoleLogin" {
+		body := makeBody(message)
+		return sendEmail(body)
 	}
 
-	return b
+	return nil
 }
 
-func parseSesUserIdentity(ui map[string]interface{}, x []string) (s string) {
-	for k, i := range ui {
-		switch v := i.(type) {
-		case string:
-			if !contains(x, k) {
-				s += k + ": " + v + "\n"
-			}
-		case map[string]interface{}:
-			s += parseSesUserIdentity(v, x)
-		default:
-			log.Printf("warning: Unknown UserIdentity value type: %T", v)
-		}
-	}
+// makeBody ... makes a text email body from a CloudWatch log event message
+func makeBody(e *ConsoleLoginEvent) (b string) {
```

Comment:
This seems like a good time to use a template or a multi-line format string

Reply:
Agreed, but I was going to to that as part of Issue #2